ephios/core/views/workinghour.py

```diff
@@ -0,0 +1,189 @@
+import datetime
+from collections import Counter
+from itertools import chain
+
+from django.contrib import messages
+from django.contrib.auth.mixins import LoginRequiredMixin
+from django.contrib.auth.models import Group
+from django.contrib.messages.views import SuccessMessageMixin
+from django.db.models import DurationField, ExpressionWrapper, F, Sum
+from django.shortcuts import redirect
+from django.urls import reverse
+from django.utils.translation import gettext as _
+from django.views.generic import DeleteView, DetailView, FormView, TemplateView, UpdateView
+from guardian.shortcuts import get_objects_for_user
+
+from ephios.core.forms.users import WorkingHourRequestForm
+from ephios.core.models import LocalParticipation, UserProfile, WorkingHours
+from ephios.extra.mixins import CustomPermissionRequiredMixin
+
+
+class WorkingHourPermissionMixin:
```

Comment:
@felixrindt I'm not super happy with this mixin, as there is a lot of similar checking across all views, but not always exactly the same. Maybe you can come up with something that is a bit cleaner?